detectable/src/main/java/com/synopsys/integration/detectable/detectables/pipenv/parse/PipfileLockExtractor.java

```diff
@@ -0,0 +1,41 @@
+package com.synopsys.integration.detectable.detectables.pipenv.parse;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.bdio.graph.DependencyGraph;
+import com.synopsys.integration.detectable.detectable.codelocation.CodeLocation;
+import com.synopsys.integration.detectable.detectables.pipenv.parse.model.PipfileLock;
+import com.synopsys.integration.detectable.detectables.pipenv.parse.model.PipfileLockDependency;
+import com.synopsys.integration.detectable.extraction.Extraction;
+
+public class PipfileLockExtractor {
+    private final Gson gson;
+    private final PipfileLockParser pipfileLockParser;
+    private final PipfileLockDependencyTransformer pipfileLockTransformer;
+
+    public PipfileLockExtractor(
+        Gson gson,
+        PipfileLockParser pipfileLockParser,
+        PipfileLockDependencyTransformer pipfileLockTransformer
+    ) {
+        this.gson = gson;
+        this.pipfileLockParser = pipfileLockParser;
+        this.pipfileLockTransformer = pipfileLockTransformer;
+    }
+
+    public Extraction extract(File pipfileLockFile, PipfileLockDetectableOptions pipfileLockDetectableOptions) throws IOException {
+        String pipfileLockText = FileUtils.readFileToString(pipfileLockFile, StandardCharsets.UTF_8);
+        PipfileLock pipfileLock = gson.fromJson(pipfileLockText, PipfileLock.class);
+        List<PipfileLockDependency> dependencies = pipfileLockParser.parse(pipfileLock, pipfileLockDetectableOptions.getDependencyTypeFilter());
```

Comment:
Since we have a dedicated parser class already, I would throw Gson in there and have the parser just accept the lock file text.

Reply:
I thought there was precedent for putting parsing/conversion done by a library in the extractor (specifically I looked at CargoExtractor, which uses the Toml library), but I guess that is a case where we don't have a dedicated parser.  I will say though that having the parser take in a string makes writing tests a lot more annoying/gross

Reply:
I like the taking in of the gson object instead of the string. Not sure why we would want it to take in the string.